edx_django_utils/security/clickjacking/README.rst

```diff
@@ -0,0 +1,28 @@
+Security Utils
+##############
+
+Common security utilities.
+
+Deprecated - Clickjacking middleware
+************************************
+
+Consider this deprecated as new projects should be using Content Security Policy headers instead.
+However, this won't be removed until no openedx project depends on it anymore, specifically, edx-platform.
+Please do not add this middleware to a project unless you have to use the ``X-Frame-Options`` header,
+and do not add it outside the openedx github organization, as it may be removed in the future.
+
+Clickjacking middleware
+***********************
+
+The preferred way to prevent clickjacking is to use the Content Security Policy headers.
+In some legacy code where those headers are not used, it is required to instead use the older
+header ``X-Frame-Options`` set either to ``DENY`` or to ``SAMEORIGIN``.
+In any case, this middleware allows you both to set the ``X-Frame-Options`` header to any recognized value -
+``DENY``, ``SAMEORIGIN``, ``ALLOW`` per django setting - but defaults to ``DENY``.
+It also allows you to override the header for specific urls defined via regex.
```

Comment:
I'm not clear. You are adding a deprecated middleware with a new feature? So this regex feature will not be needed in some future time?

Reply:
Okay let me try to explain my intention. Probably there's a better way to do this.

From a security perspective, we prefer CSP over X-Frame-Options for new or flexible projects. That means that it's okay to use X-Frame-Options but only if the project already has them and it's more work than it's worth to change them. That would be way outside the scope of my task for edx-platform - edx-platform still using the `X-Frame-Options` header.

The new middleware is important because we currently can't support SCORM xblock without setting the header to `SAMEORIGIN`, but I did not want to blindly do this for all of edx-platform. So I wrote this code to allow us to change the header for specific url patterns.

I wanted to mark this as deprecated because on the one hand we need it and it's not worth the effort to change the whole security headers thing on edx-platform right now, but I want to warn people off from using this middleware together with the outdated header in new projects.

Do you have a better suggestion for this?

Reply:
I may have more thoughts, but a quick question is whether this is for edx-platform only, and if it should live there? That would reduce scope of introducing deprecated-from-the-start code.